src/callbacks_step/summary.jl

```diff
@@ -14,42 +22,107 @@ function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condi
 end
 
 
+function format_key_value_line(key::AbstractString, value::AbstractString, key_width, total_width;
+                               guide='…', filler='…', prefix="│ ", suffix=" │")
+  @assert key_width < total_width
+  line  = prefix
+  squeezed_key = squeeze(key, key_width, filler=filler)
+  line *= squeezed_key
+  line *= ": "
+  short = key_width - length(squeezed_key)
+  if short <= 1
+    line *= " "
+  else
+    line *= guide^(short-1) * " "
+  end
+  value_width = total_width - length(prefix) - length(suffix) - key_width - 2
+  squeezed_value = squeeze(value, value_width, filler=filler)
+  line *= squeezed_value
+  short = value_width - length(squeezed_value)
+  line *= " "^short
+  line *= suffix
+
+  @assert length(line) == total_width "should not happen: algorithm error!"
+
+  return line
+end
+format_key_value_line(key, value, args...; kwargs...) = format_key_value_line(string(key), string(value), args...; kwargs...)
+
+function squeeze(message, max_width; filler='…')
+  @assert max_width >= 3 "squeezing works only for a minimum `max_width` of 3"
+
+  length(message) <= max_width && return message
+
+  keep_front = div(max_width, 2)
+  keep_back = div(max_width, 2) - (isodd(max_width) ? 0 : 1)
+  squeezed = message[begin:keep_front] * filler * message[end-keep_back+1:end]
```

Comment:
```suggestion
  squeezed = message[begin:begin+keep_front-1] * filler * message[end-keep_back+1:end]
```
Otherwise, you assume 1-based indexing here, don't you?